jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -128,21 +129,42 @@ object XGBoost extends Serializable {
       }
       rabitEnv.put("DMLC_TASK_ID", TaskContext.getPartitionId().toString)
       Rabit.init(rabitEnv)
-      val trainingMatrix = new DMatrix(
-        fromDenseToSparseLabeledPoints(trainingPoints, missing), cacheFileName)
+      val trainTestRatio = params.get("trainTestRatio").map(_.toString.toDouble).getOrElse(1.0)
+      // In the worst-case this would store [[trainTestRatio]] of points
+      // buffered in memory.
+      val r = params.get("seed")
+          .map(v => new Random(v.toString.toLong))
+          .getOrElse(new Random())
+      val (trainPoints, testPoints) = fromDenseToSparseLabeledPoints(labeledPoints, missing)
+          .partition(_ => r.nextDouble() <= trainTestRatio)
+      val trainMatrix = new DMatrix(trainPoints, cacheFileName)
+      val testMatrix = new DMatrix(testPoints, cacheFileName)
```

Comment:
what will happen if the user sets trainTestRatio to 1.0, what's contained in testMatrix, and is it still safe to call testMatrix.delete()?

Reply:
The `testMatrix` would be empty (this is the case for all of the tests already), and I'd expect it to produce no files, effectively making `.delete` a noop.